tests/e2e/cleanup-ocp.sh

```diff
@@ -0,0 +1,41 @@
+#!/bin/bash
+
+# Copyright Istio Authors
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+set -eux -o pipefail
+
+# Clean up leftover artifacts from e2e.ocp tests
+echo "Cleaning up leftover artifacts from e2e.ocp tests"
+
+# Check if KUBECONFIG is set
+if [ -z "${KUBECONFIG}" ]; then
+    echo "KUBECONFIG is not set. kubectl will not be able to connect to the cluster. Exiting."
+    exit 1
+fi
+
+# Clean up cluster-level resources that may be left over from e2e tests
```

Comment:
It might be a good practice if we do include the following steps here (to give the operator a chance to perform any cleanup operations).
1. verify if Istio is running, uninstall it.
2. verify if Sail/OSSM Operator is installed and delete it.